include/uprotocol-cpp/uri/builder/BuildUAuthority.h

```diff
@@ -0,0 +1,133 @@
+/*
+ 
+ * Copyright (c) 2024 General Motors GTO LLC
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * 
+ * SPDX-FileType: SOURCE
+ * SPDX-FileCopyrightText: 2023 General Motors GTO LLC
+ * SPDX-License-Identifier: Apache-2.0
+ 
+ 
+ */
+
+#ifndef UPROTOCOL_CPP_BUILDUAUTHORITY_H
+#define UPROTOCOL_CPP_BUILDUAUTHORITY_H
+
+#include <algorithm>
+#include <cctype>
+#include <optional>
+#include <string>
+#include <array>
+#include <string_view>
+#include <arpa/inet.h>
+#include <spdlog/spdlog.h>
+#include "../tools/Utils.h"
+#include "uprotocol/uri.pb.h"
+
+namespace uprotocol::uri {
+    class BuildUAuthority {
+        uprotocol::v1::UAuthority authority_;
+    public:
+        BuildUAuthority() { authority_.Clear(); }
+    
+        auto setName(const std::string &name) -> BuildUAuthority & {
+            if (authority_.has_name() && !authority_.name().empty()) {
+                spdlog::error("UAutority already has a remote set. Ignoring setName()");
+                return *this;
+            }
+            if (isBlank(name)) {
+                spdlog::error("UAutority name is blank. Ignoring setName()");
+                return *this;
+            } else {
+                auto tmp  = name;
+                std::transform(tmp.begin(), tmp.end(), tmp.begin(),
```

Comment:
What is this for?

Reply:
since we use lower case and user may continue using his setting for some reason, we must copy it as lower case.